.github/workflows/java-ecs-canary.yml

```
@@ -0,0 +1,27 @@
+## Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+## SPDX-License-Identifier: Apache-2.0
+
+## This workflow aims to run the Application Signals Java end-to-end tests as a canary to
+## test the artifacts for Application Signals enablement. It will deploy a sample app onto an ECS cluster,
+## call the APIs, and validate the generated telemetry, including logs, metrics, and traces.
+name: Java ECS Enablement Canary Testing
+on:
+  schedule:
+    - cron: '*/15 * * * *' # run the workflow every 15 minutes
+  workflow_dispatch: # be able to run the workflow on demand
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  ecs:
+    strategy:
+      fail-fast: false
+      matrix:
+        aws-region: ['us-east-1']
+    uses: ./.github/workflows/java-ecs-retry.yml
+    secrets: inherit
+    with:
+      aws-region: ${{ matrix.aws-region }}
+      caller-workflow-name: 'appsignals-e2e-java-ecs-canary-test'
```

If you intend on testing in IAD alone, then you don't need a matrix.
```suggestion
  ecs:
    uses: ./.github/workflows/java-ecs-retry.yml
    secrets: inherit
    with:
      aws-region: 'us-east-1'
      caller-workflow-name: 'appsignals-e2e-java-ecs-canary-test'
```

---

For now, we only want to test in IAD, but I think it will be better to use matrix, so that it will be convenient to add more regions if we want to test in the future.